File: app/src/main/java/com/example/schedule/SQL/SQLScheduleReader.java
package com.example.schedule.SQL;

import android.content.Context;
import android.database.Cursor;
import android.database.sqlite.SQLiteDatabase;

public class SQLScheduleReader extends SQLManager {

    private SQLiteDatabase sqLiteDatabase;

    public SQLScheduleReader(Context context, String tableName, int version) {
        super(context, tableName, null, version);
        sqLiteDatabase = getReadableDatabase();
    }

    public Cursor getScheduleByDay(String[] columns, int dayOfWeek, int weekStatus) {

        String selection = SQLManager.DAY_OF_WEEK + " = ? AND (" +
                SQLManager.BOTH_NUMERATOR_DIVIDER + " = ? OR " +
                SQLManager.BOTH_NUMERATOR_DIVIDER + " = ? )";

        String[] selectionArgs = new String[]{
                String.valueOf(dayOfWeek),
                String.valueOf(SQLManager.BOTH),
                String.valueOf(weekStatus)};

        return sqLiteDatabase.query(getDatabaseName(), columns,
                selection, selectionArgs,
                null, null, null);
    }

}

File: app/src/main/java/com/example/schedule/SQL/SQLScheduleReader.kt
package com.example.schedule.SQL;

import android.content.Context;
import android.database.Cursor;
import android.database.sqlite.SQLiteDatabase;

class SQLScheduleReader(context: Context,
                        tableName: String,
                        val version: Int)
    : SQLManager(context, tableName,
        null, version) {

    private var sqLiteDatabase: SQLiteDatabase = readableDatabase

    fun getScheduleByDay(columns: Array<String>, dayOfWeek: Int): Cursor {
        if (dayOfWeek !in 1..7) throw IllegalArgumentException("DayOfWeek $dayOfWeek must be in 1..7")

        val selection = "$DAY_OF_WEEK = ? "

        val selectionArgs = arrayOf(dayOfWeek.toString())

        return sqLiteDatabase.query(databaseName, columns,
                selection, selectionArgs,
                null, null, null)
    }
}
